src/templates/QuickstartDetails.js

```diff
@@ -26,6 +26,62 @@ const layoutContentSpacing = css`
   margin: auto;
 `;
 
+const moveToBack = (componentsAndCounts) => {
+  const n = componentsAndCounts.length;
+  const newOrder = [];
+
+  // add components with content
+  // to the beginning of array
+  let i = 0;
+  while (i < n) {
+    if (componentsAndCounts[i].count !== 0) {
+      // Push element to end of array.
+      // This keeps the same order if all components
+      // have a length of 0.
+      newOrder.push(componentsAndCounts.splice(i, 1)[0]);
+    }
+    i++;
+  }
+
+  // add components without content
+  // to the end of array
+  while (componentsAndCounts.length > 0) {
+    // push the remaining elements in order to
+    // new ordered array
+    newOrder.push(componentsAndCounts.shift());
+  }
+  return newOrder;
+};
+
+const sortOrderedQuickstartComponents = (quickstart) => {
+  // get length of all components
+  const dashboardLength = quickstart.dashboards?.length ?? 0;
+  const alertLength = quickstart.alerts?.length ?? 0;
+
+  // we use documentation for datasources at the moment
+  const dataSourceLength = quickstart.documentation?.length ?? 0;
+
+  // sort by length
+  const componentsAndCounts = [
+    {
+      component: Dashboards,
+      count: dashboardLength,
+    },
+    { component: Alerts, count: alertLength },
+    { component: DataSources, count: dataSourceLength },
+  ];
+
+  // Check to see if we need to move empty components to
+  // the end of the array
+  const hasEmptyComponent = componentsAndCounts.filter(
+    (component) => component.count
+  );
+
+  return hasEmptyComponent.length > 0
+    ? moveToBack(componentsAndCounts)
+    : componentsAndCounts;
+};
```

Comment:
I could be wrong here, but I think we could replace all this with
```suggestion
  // Check to see if we need to move empty components to
  // the end of the array
  return componentsAndCounts.sort((a, b) => {
    if (a.length < 1) {
      return -1;
    } else if (b.length < 1) {
      return 1;
    } else {
      return 0;
    }
  })
};
```

Reply:
Yup! i was definitely missing something. Only correct to this is 

```jsx
if (a.count < 1) {
      return 1;
    } else if (b.count < 1) {
      return -1;
```
in order to keep it in the correct order 👍🏻